src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -400,3 +410,285 @@ function getBucketHashFromItemHash(
   const def = defs.InventoryItem.get(itemHash);
   return def && ('bucketTypeHash' in def ? def.bucketTypeHash : def.inventory?.bucketTypeHash);
 }
+
+/**
+ * Remove all Loadout Optimizer parameters from a loadout. This leaves things like mods and fashion in place.
+ */
+export function clearLoadoutParameters(): LoadoutUpdateFunction {
+  return produce((draft) => {
+    if (draft.parameters) {
+      delete draft.parameters.assumeArmorMasterwork;
+      delete draft.parameters.exoticArmorHash;
+      delete draft.parameters.lockArmorEnergyType;
+      delete draft.parameters.query;
+      delete draft.parameters.statConstraints;
+      delete draft.parameters.upgradeSpendTier;
+      delete draft.parameters.autoStatMods;
+    }
+  });
+}
+
+/** Remove the current subclass from the loadout. */
+export function clearSubclass(defs: D2ManifestDefinitions): LoadoutUpdateFunction {
+  return (loadout) => {
```

Comment:
Maybe we should use immer on all these things? Might be overkill for some but consistency is nice also 🤔

Reply:
Yeah I just didn't use it where I didn't think it added a lot